db/crud.go

```diff
@@ -1899,6 +1899,10 @@ func (db *DatabaseCollectionWithUser) updateAndReturnDoc(ctx context.Context, do
 	if db.UseXattrs() || upgradeInProgress {
 		var casOut uint64
 		// Update the document, storing metadata in extended attribute
+		if opts == nil {
+			opts = &sgbucket.MutateInOptions{}
```

Comment:
is this `DefaultMutateInOpts()` ?

Reply:
It's possible that the caller has already set MutateInOptions.PreserveExpiry, so just initializing here and then adding the MacroExpansion (for either case) below.